src/commands/synthetics/tunnel.ts

```diff
@@ -223,7 +230,7 @@ export class Tunnel {
   }
 
   private async getConnectionInfo() {
-    const [rawConnectionInfo] = (await once(this.ws!, 'message')) as (string | Buffer)[]
+    const rawConnectionInfo = await this.firstMessagePromise
```

Comment:
Rather than storing the promise here, can't you store in `this.ws` and expose a `await this.ws.waitForFirstMessage`?